src/App.css

```diff
@@ -0,0 +1,88 @@
+.main-container {
```

Comment:
> I struggled at the point where I should decide how to organise my styles component based.  Should I make everything a component? should every component has its own styles? 

In React projects is very rare to use CSS like this — global CSS.

The main problem with this CSS in big projects you'll have a clash of class names, which can create unwanted bugs and make it harder to escalate CSS.

So, an alternative to it is to use CSS Modules.

I'd highly recommend you to give it a try, in this MR or in the next one.

- [How to setup in Create React App](https://create-react-app.dev/docs/adding-a-css-modules-stylesheet/)
- [Youtube tutorial](https://www.youtube.com/watch?v=ZawAwPYrxGA) explaining the benefits

The idea is for the CSS to be right next to their component. (as you did with button, but applied to all components)

Globally (`styles.css`) the only styles that exist are [reset styles](https://meyerweb.com/eric/tools/css/reset/) and styles to global selectors.